src/platform/site-wide/user-nav/components/Typeahead.jsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import environment from 'platform/utilities/environment';
+
+export const isTypeaheadEnabled =
+  !environment.isProduction() && document.location.pathname === '/';
+
+export const typeaheadListId = 'onsite-search-typeahead';
+
+export default class Typeahead extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      suggestions: [],
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    const inputChanged = prevProps.userInput !== this.props.userInput;
+    if (inputChanged) {
+      this.getSuggestions();
+    }
+  }
+
+  async getSuggestions() {
+    const input = this.props.userInput;
+    if (input?.length <= 2) return;
+
+    const encodedInput = encodeURIComponent(input);
+    const response = await fetch(
+      `https://search.usa.gov/sayt?=&name=va&q=${encodedInput}`,
+    );
+
+    const suggestions = await response.json();
+
+    this.setState({ suggestions });
+  }
+
+  render() {
+    return (
```

Comment:
Should we escape early with a `return null` if `this.props.userInput` is empty?

Reply:
Hm I was thinking it would be okay for `<datalist>` to have no children, but you might be right...I'll check the docs

Reply:
I think this is okay to be empty because the [docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist) say `zero or more <option> elements.`